gcsfs/core.py

```diff
@@ -837,22 +837,25 @@ async def _mkdir(
         location=None,
         create_parents=True,
         enable_versioning=False,
+        enable_object_retention=False,
```

Comment:
Added explicit handling for `enableObjectRetention` because it is the only other arg not passed through the request body, and `iamConfiguration` because it requires removing `acl` args to prevent a 400. Anything else we want can now be passed through `kwargs` to the request body.

Reply:
+1